solidity/dashboard/src/components/ClosableContainer.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect } from "react"
+import * as Icons from "./Icons"
+
+const ClosableContainer = ({ hide = false, className, children }) => {
+  const [showComponent, setShowComponent] = React.useState(true)
+
+  useEffect(() => {
+    setShowComponent(!hide)
+  }, [hide])
+
+  const hideComponent = () => {
+    setShowComponent(false)
+  }
```

Comment:
Probably we can extract these lines to a custom hook and use it in other components.

Reply:
I extracted these lines to a custom hook like you said. I would appreciate if you could look into this and tell me if this is what you meant https://github.com/keep-network/keep-core/pull/2250/commits/4702792955db31d34f53ed1032ecd74d2a14f736 🙏